provider/service.py

```diff
@@ -145,17 +152,48 @@ def stopChangesFeed(self):
     def createAndRunConsumer(self, doc):
         triggerFQN = doc['_id']
 
+        if doc['isMessageHub']:
+            retries = 0
+            max_retries = 5
+            url = '{}/admin/topics'.format(doc['kafka_admin_url'])
+            username = doc['username']
+            password = doc['password']
+            topic = doc['topic']
+
+            while retries < max_retries:
+                headers = {'X-Auth-Token': '{}{}'.format(username, password)}
+                response = self.session.get(url, headers=headers, stream=False, timeout=10.0, verify=check_ssl)
+                status_code = response.status_code
+
+                if status_code == 200:
+                    topics = json.loads(response.content)
+
+                    if topic in [t['name'] for t in topics]:
+                        break
+                    else:
+                        logging.warn('Topic {} for trigger {} no longer exists. This consumer will not be created and trigger will be disabled'.format(topic, triggerFQN))
+                        reason = 'Topic does not exist. You must create the topic first: {}.'.format(url, topic)
+                        self.database.disableTrigger(triggerFQN, 404, reason)
+                        return
+                elif status_code == 403:
+                    logging.warn('[{}] Invalid authKey.  This consumer will not be created and trigger will be disabled'.format(triggerFQN))
+                    reason = 'Invalid authKey. {} returned 403 using authKey {}{}.'.format(url, username, password)
+                    self.database.disableTrigger(triggerFQN, 403, reason)
+                    return
+                else:
+                    sleepyTime = pow(2, retries)
+                    logging.info('Received status code {} for {} while validating authKey. Retrying in {} second(s)'.format(status_code, triggerFQN, sleepyTime))
+                    time.sleep(sleepyTime)
+                    retries += 1
+
         # Create a representation for this trigger, even if it is disabled
         # This allows it to appear in /health as well as allow it to be deleted
         # Creating this object is lightweight and does not initialize any connections
         consumer = Consumer(triggerFQN, doc)
         self.consumers.addConsumerForTrigger(triggerFQN, consumer)
 
-        if self.__isTriggerDocActive(doc):
-            logging.info('[{}] Trigger was determined to be active, starting...'.format(triggerFQN))
-            consumer.start()
-        else:
-            logging.info('[{}] Trigger was determined to be disabled, not starting...'.format(triggerFQN))
+        logging.info('[{}] Starting consumer...'.format(triggerFQN))
```

Comment:
Do these changes affect the other occurrence of the call to `__isTriggerDocActive` in this file?

Reply:
yes. currently we detect a change in the changes feed and check whether that consumer is in our in-memory cache. this is still the case, however, since we will not be creating disabled consumers. the only time we'll hit the code path where we are going from inactive -> active **and**

Reply:
we **do** have that consumer in memory is when we disable a trigger for reasons like client errors. doing that is something that may likely change in the future. so that whole logic would need reworking